### PR TITLE
[full-ci] [tests-only] Added tests for auto-sync disabled

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
Added Scenarios:
- `Scenario: user who is also a member of group lists the file shared with them`
- `Scenario: user who is also a member of group lists the folder shared with them`

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/8318

## How Has This Been Tested?

- Locally
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 1]
```diff
@@ -1617,3 +1617,285 @@ Feature: an user gets the resources shared to them
...
+                "@client.synchronize",
+                "createdBy",
+                "eTag",
+                "file",
+                "id",
+                "lastModifiedDateTime",
+                "name",
+                "parentReference",
+                "remoteItem",
+                "size"
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1668]: Should all the properties in the required field be checked? If not maybe they can be skipped.

**Comment by nabim777** [comment]
Scenario related to `auto-sync disable` is moved to another new file `sharedWithMeSyncDisabled.feature` please move there https://github.com/owncloud/ocis/pull/8443

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 2]
```diff
@@ -1617,3 +1617,285 @@ Feature: an user gets the resources shared to them
         }
       }
       """
+
+
+  Scenario: user who is also a member of group lists the file shared with them when auto-sync is disabled
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1622]:
```suggestion
  Scenario: user who is also a member of group, lists the file shared with them when auto-sync is disabled
```

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature (hunk 1 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1668]: imo all the mentioned properties should be checked here

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature [hunk 3]
```diff
@@ -1617,3 +1617,285 @@ Feature: an user gets the resources shared to them
...
+                                "required": ["group"],
+                                "properties": {
+                                  "group": {
+                                    "type": "object",
+                                    "required": ["displayName", "id"],
+                                    "properties": {
+                                      "displayName": {
+                                        "type": "string",
+                                        "enum": ["grp1"]
+                                      }
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1744]: why oneOf don't we have fix behavior?

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature (hunk 1 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature:1668]: I didn't see these properties being checked, so mentioned it. So, we need to add checks for these properties as well.

Comment (grgprarup) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: The complete response test is covered in previous tests for sync disabled state. IMO, we don't need to validate every item in this test coverage. The presence of items in response can be checked through `required`. When the auto sync is disabled the only change made in response is `"@client.synchronize"`

File: tests/acceptance/features/apiSharingNg/sharedWithMe.feature (hunk 3 above)
Comment (grgprarup) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: The order of the array items is random. I think adding that won't affect anything.

Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: Better if you avoid using oneOf if possible if not then we have to use it by checking false positive cases. 
did you try allOf ?

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: Please add the min and max count:
```diff
  "type": "array",
+ "minItems": 2,
+ "maxItems": 2,
```
this should make the assertion little bit more strict

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: we must use `minItems` adn `maxItems` for `oneOf` as well

Comment (SagarGi) [reply tests/acceptance/features/apiSharingNg/sharedWithMe.feature]: @saw-jan yes agreed but @nirajacharya2 have a thorough check all of it.

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 4]
```diff
@@ -1034,3 +1034,285 @@ Feature: listing sharedWithMe when auto-sync is disabled
         }
       }
       """
+
+
+  Scenario: user who is also a member of group lists the file shared with them when auto-sync is disabled
+    Given user "Carol" has been created with default attributes and without skeleton files
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:1040]: is user carol needed?

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 5]
```diff
@@ -1034,3 +1034,285 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+    And group "grp1" has been created
+    And the following users have been added to the following groups
+      | username | groupname |
+      | Brian    | grp1      |
+      | Carol    | grp1      |
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:1045]: also adding carol to the group?

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**